### PR TITLE
Implement URL filter and pagination when using PDO

### LINKED ISSUE #303: Support basic url filter with PDO backend
The workflow for Wikimedia's XHGui is to instrument a page load, and then get a link to `https://xhgui/?url=<requestID>` ([example](https://performance.wikimedia.org/xhgui/?url=3687efbd-ccda-4859-8dbe-0ab4e65a98b1)), which yields one result, and is the one the user is interested in.

With the new PDO backend, it seems none of the search options are implemented yet, so this currently leads to all results. This is particularly challenging since a single page load tends to spawn multiple request and thus it'd require the developer to look at the request ID that XHGui is searching for and compare it against the top 20 results or so by eye.

There's more to be done in this area, but I'm filing this one separately so that perhaps Wikimedia can help implement this one, and/or to report back if we find another way :)

### PR DESCRIPTION
**Pull request by dpifke**
Fixes #303.

### CODE REVIEW

**Comment by glensc** [comment]
@dpifke does this solve any open issue? please indicate so in PR body if that is so.

File: src/Xhgui/Searcher/Pdo.php [hunk 1]
```diff
@@ -146,10 +146,28 @@ public function getAll($options = array())
...
         $sort = $options['sort'];
         $direction = $options['direction'];
-        $page = $options['page'];
-        $perPage = $options['perPage'];
+        $page = (int)$options['page'];
+        if ($page < 1) {
+            $page = 1;
+        }
+        $perPage = (int)$options['perPage'];
+        $url = $options['conditions']['url'] ?? "";
```
Comment (glensc) [inline src/Xhgui/Searcher/Pdo.php:154]: this syntax is not supported by php 5.6, which is this project min requirement:
- https://github.com/perftools/xhgui/blob/0.11.0/composer.json#L11

File: src/Xhgui/Searcher/Pdo.php [hunk 2]
```diff
@@ -146,10 +146,28 @@ public function getAll($options = array())
...
 
-        $stmt = $this->pdo->query("
+        $stmt = $this->pdo->prepare("
+          SELECT COUNT(*) AS count
+          FROM {$this->table}
+          WHERE simple_url LIKE :url
+        ");
+        $stmt->execute(['url' => '%'.$url.'%']);
+        $totalRows = $stmt->fetch(PDO::FETCH_ASSOC)['count'];
```
Comment (glensc) [inline src/Xhgui/Searcher/Pdo.php:162]: seems you want [fetchColumn](https://www.php.net/manual/en/pdostatement.fetchcolumn.php) here.

**Comment by glensc** [comment]
you shouldn't put two changesets to one PR. the one suitable for merge extracted as separate PR: https://github.com/perftools/xhgui/pull/306 (rebase once it's merged), the other changeset requires changes before merge.

**Comment by dpifke** [comment]
> you shouldn't put two changesets to one PR. the one suitable for merge extracted as separate PR: #306 (rebase once it's merged), the other changeset requires changes before merge.

My understanding is that there isn't a way to make one PR in Github dependent on another.  The fix for PDO argument ordering was required for this to function at all.

**Comment by glensc** [comment]
> My understanding is that there isn't a way to make one PR in Github dependent on another. The fix for PDO argument ordering was required for this to function at all.

have you tried changing "target branch"? works in gitlab. but you can always indicate dependency in merge request body and use "work in progress" state to prevent early merge.

**Pull request merged**